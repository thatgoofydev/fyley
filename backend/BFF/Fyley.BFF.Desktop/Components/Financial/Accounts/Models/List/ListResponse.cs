using System;
using JetBrains.Annotations;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Fyley.BFF.Desktop.Components.Financial.Accounts.Models.List
{
    public class ListResponse
    {
        [UsedImplicitly] public AccountViewModel[] Accounts { get; }
        
        public ListResponse(AccountViewModel[] accounts)
        {
            Accounts = accounts;
        }

        public class AccountViewModel
        {
            public Guid AccountId { get; set; }
            public string Name { get; set; }
            public string Description { get; set; }
            public AccountNumberType AccountNumberType { get; set; }
            public string AccountNumber { get; set; }
        }
        
        [JsonConverter(typeof(StringEnumConverter))]
        public enum AccountNumberType
        {
            Other = 1,
            Iban = 2
        }
    }
}