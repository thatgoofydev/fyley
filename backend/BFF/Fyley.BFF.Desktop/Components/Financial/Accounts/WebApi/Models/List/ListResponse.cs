using System;
using JetBrains.Annotations;

namespace Fyley.BFF.Desktop.Components.Financial.Accounts.WebApi.Models.List
{
    public class ListResponse
    {
        [UsedImplicitly] public AccountViewModel[] Accounts { get; }
        
        public ListResponse(AccountViewModel[] accounts)
        {
            Accounts = accounts;
        }

        public class AccountViewModel
        {
            public Guid AccountId { get; set; }
            public string Name { get; set; }
            public string Description { get; set; }
            public string AccountNumberType { get; set; }
            public string AccountNumber { get; set; }
        }
    }
}