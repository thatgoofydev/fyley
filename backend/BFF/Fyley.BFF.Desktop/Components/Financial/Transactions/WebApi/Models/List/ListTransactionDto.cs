namespace Fyley.BFF.Desktop.Components.Financial.Transactions.WebApi.Models.List
{
    public class ListTransactionDto
    {
        public string Payor { get; set; }
        public string Payee { get; set; }
        public decimal Amount { get; set; }
        public string Reference { get; set; }
        public string OccuredOn { get; set; }
        public string LoggedOn { get; set; }
    }
}