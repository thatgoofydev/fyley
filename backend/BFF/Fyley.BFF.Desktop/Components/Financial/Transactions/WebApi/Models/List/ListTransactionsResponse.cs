namespace Fyley.BFF.Desktop.Components.Financial.Transactions.WebApi.Models.List
{
    public class ListTransactionsResponse
    {
        public ListTransactionDto[] Transactions { get; set; }
    }
}