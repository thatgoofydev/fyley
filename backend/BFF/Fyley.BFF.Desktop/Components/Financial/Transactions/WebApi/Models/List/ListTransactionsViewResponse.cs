using System;

namespace Fyley.BFF.Desktop.Components.Financial.Transactions.WebApi.Models.List
{
    public class ListTransactionsViewResponse
    {
        public TransactionDto[] Transactions { get; set; }
        
        public class TransactionDto
        {
            public Guid TransactionId { get; set; }
            public AccountDetails Payor { get; set; }
            public AccountDetails Payee { get; set; }
            public decimal Amount { get; set; }
            public string Reference { get; set; }
            public string OccuredOn { get; set; }

            public class AccountDetails
            {
                public string Name { get; set; }
                public string AccountNumber { get; set; }
            }
        }
    }
}