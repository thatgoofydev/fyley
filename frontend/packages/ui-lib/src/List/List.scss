@import "src/base-styling/vars";

.list {
  .list-item {
    padding: 10px 20px;
    border: $list-item-border;
    border-radius: $border-radius;
    margin-bottom: 6px;
    background-color: $list-item-bg-color;

    &:hover {
      background-color: darken($list-item-bg-color, 3%);
      cursor: pointer;
      text-decoration: underline;
    }
  }
}

.list-compact {
  .list-item {
    padding: 10px 20px;
    border-top: none;
    border-radius: 0;
    margin-bottom: 0;

    &:first-child {
      border-top: $list-item-border;
      border-top-left-radius: $list-item-border-radius;
      border-top-right-radius: $list-item-border-radius;
    }

    &:last-child {
      border-bottom-left-radius: $list-item-border-radius;
      border-bottom-right-radius: $list-item-border-radius;
    }
  }
}

