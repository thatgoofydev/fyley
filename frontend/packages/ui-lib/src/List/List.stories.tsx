import React from "react";
import { Meta, Story } from "@storybook/react";
import { List, ListItem, IProps } from "./List";

export default {
  title: "General/List",
  component: List,
  subcomponents: {
    ListItem
  }
} as Meta;

export const TextList: Story<IProps> = args => {
  return (
    <List {...args}>
      <ListItem>Item 1</ListItem>
      <ListItem>Item 2</ListItem>
      <ListItem>Item 3</ListItem>
      <ListItem>Item 4</ListItem>
    </List>
  );
};
TextList.args = {
  type: "normal"
};
