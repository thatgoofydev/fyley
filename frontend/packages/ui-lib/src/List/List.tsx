import React, { FunctionComponent } from "react";
import classNames from "classnames";
import "./List.scss";

export interface IProps {
  type?: "normal" | "compact" | undefined;
}

export const List: FunctionComponent<IProps> = ({ type = "normal", children }) => {
  const classes = classNames("list", "list-" + type);
  return <div className={classes}>{children}</div>;
};

export const ListItem: FunctionComponent = ({ children }) => {
  return <div className="list-item">{children}</div>;
};
