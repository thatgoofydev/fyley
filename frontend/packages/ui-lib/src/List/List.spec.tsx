import React from "react";
import { render, cleanup } from "@testing-library/react";
import { List, ListItem } from "./List";

afterEach(cleanup);

describe("Loader", function () {
  test("should render with normal", () => {
    const rendered = render(
      <List>
        <ListItem>Item 1</ListItem>
        <ListItem>Item 2</ListItem>
        <ListItem>Item 3</ListItem>
        <ListItem>Item 4</ListItem>
      </List>
    );

    expect(rendered).toMatchSnapshot();
  });

  test("should render without spacing when type=compact", () => {
    const rendered = render(
      <List type="compact">
        <ListItem>Item 1</ListItem>
        <ListItem>Item 2</ListItem>
        <ListItem>Item 3</ListItem>
        <ListItem>Item 4</ListItem>
      </List>
    );

    expect(rendered).toMatchSnapshot();
  });
});
