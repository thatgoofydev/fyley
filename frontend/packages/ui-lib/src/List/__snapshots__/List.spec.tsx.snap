// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render with normal 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="list list-normal"
      >
        <div
          class="list-item"
        >
          Item 1
        </div>
        <div
          class="list-item"
        >
          Item 2
        </div>
        <div
          class="list-item"
        >
          Item 3
        </div>
        <div
          class="list-item"
        >
          Item 4
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="list list-normal"
    >
      <div
        class="list-item"
      >
        Item 1
      </div>
      <div
        class="list-item"
      >
        Item 2
      </div>
      <div
        class="list-item"
      >
        Item 3
      </div>
      <div
        class="list-item"
      >
        Item 4
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Loader should render without spacing when type=compact 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="list list-compact"
      >
        <div
          class="list-item"
        >
          Item 1
        </div>
        <div
          class="list-item"
        >
          Item 2
        </div>
        <div
          class="list-item"
        >
          Item 3
        </div>
        <div
          class="list-item"
        >
          Item 4
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="list list-compact"
    >
      <div
        class="list-item"
      >
        Item 1
      </div>
      <div
        class="list-item"
      >
        Item 2
      </div>
      <div
        class="list-item"
      >
        Item 3
      </div>
      <div
        class="list-item"
      >
        Item 4
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
